test/integration/project-template/pre_commit_test.py

```diff
@@ -0,0 +1,68 @@
+import subprocess
+from pathlib import Path
+
+import pytest
+
+
+@pytest.fixture(scope="module")
+def poetry_path() -> str:
+    result = subprocess.run(["which", "poetry"], capture_output=True, text=True)
+    poetry_path = result.stdout.strip()
+    return poetry_path
+
+
+@pytest.fixture(scope="module")
+def git_path() -> str:
+    result = subprocess.run(["which", "git"], capture_output=True, text=True)
+    git_path = result.stdout.strip()
+    return git_path
+
+
+@pytest.fixture(scope="module")
+def run_command(poetry_path, git_path, new_project):
+    def _run_command_fixture(command, **kwargs):
+        defaults = {
+            "capture_output": True,
+            "check": True,
+            "cwd": new_project,
+            "env": {"PATH": f"{Path(git_path).parent}:{Path(poetry_path).parent}"},
+            "text": True,
+
+        }
+        config = {**defaults, **kwargs}
+
+        return subprocess.run(command, **config)
+
+    return _run_command_fixture
+
+
+@pytest.fixture(scope="module")
+def pre_commit(run_command, new_project, poetry_path):
+    run_command(command=["git", "init"])
+    run_command([poetry_path, "add", "pre-commit"])
```

Comment:
could add instead as a dev dependency to the cookiecutter templates `pyproject.toml`. thoughts?

Reply:
ah, I thought, pre-commit was a default part of ptb, so I thought it would already somehow be in a dependency. Either as a direct dependency of ptb and as such a transitive dependency of project is using the ptb.